File: cpp/Program/Turtle.cpp
#include "Turtle.h"

Turtle::Turtle(string n, double l, string h, string d, string s)
    : Reptile(n, l, h, d), shellType(s) {}

void Turtle::display() {
    cout << "Turtle: " << name
         << ", Shell: " << shellType
         << ", Habitat: " << habitat << endl;
}

File: java/program/Turtle.java
public class Turtle extends Reptile {
    private String shellType;

    public Turtle(String name, double length, String habitat, String diet, String shellType) {
        super(name, length, habitat, diet);
        this.shellType = shellType;
    }

    @Override
    public void display() {
        System.out.println("Turtle: " + name +
                           ", Shell: " + shellType +
                           ", Habitat: " + habitat);
    }
}
